azure/README.md

```diff
@@ -0,0 +1,104 @@
+
+4. #### Generate Storage and Docker Image secrets
+
+This step requires user blob storage account and docker image details.
+Instructions for accessing storage account keys can be found at [portal](https://docs.microsoft.com/en-us/azure/storage/common/storage-account-keys-manage), [CLI](https://docs.microsoft.com/en-us/cli/azure/storage/account/keys)  
+
+##### Generate Storage secret
+```
+python petctl.py storage_secret --account_name <storage account name> 
+                                --account_key "<storage account key>" 
+```
+##### Generate Docker image secret
+```
+python petctl.py docker_secret --server <docker server> 
+                               --username <docker username> 
+                               --password <docker password>
+                               --image_name <docker image name>
+```
+
+Training job file is updated to mount users storage account onto worker instances and apply the user provided docker image.
+Base docker image to run Pytorch Elastic on Azure is at [Dockerfile](config/Dockerfile). Instructions on publishing docker image to  AzureContainer registry can be found at [ACR](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-get-started-docker-cli).
+Docker image secret generation can be skipped for running [imagenet](../../examples/imagenet/main.py) example as job specs yaml is already populated with public AzureML image with pytorch elastic support.
+
+5. #### Start your training job
+
+Submit the training job.
+```
+python petctl.py run_job
+```
+To run the provided imagenet example, training script and data can be uploaded to Azure blob storage by running
+```
+python petctl.py --upload_storage --source_path ../examples/imagenet/main.py
+                                  --account_name <storage account name>
+                                  --container_name code
+                                  --sas_token <sas token for blob storage>
+```
+```
+python petctl.py --upload_storage --source_path <path to imagenet train folder>
+                                  --account_name <storage account name>
+                                  --container_name data
+                                  --sas_token <sas token for blob storage>
+```
+
+6. #### Check status of your job
+```
+python petctl.py check_status
+```
+7. #### Scale worker instances
+```
+python petctl.py scale --rg "<resource_group>"
+                       --location "<location>"
+                       --subscription_id <subscription_id>
+                       --client_id <service principal client id>
+                       --client_secret <service principal client secret>
+                       --new_node_count <worker instances count>    
+```
+(Here subscription id and resource group is the one setup in [Prerequisites](#Prerequisites) #3.
+
+8. #### Delete resources
+````
+python petctl.py delete_resources
```

Comment:
Can be just replaced with `az group delete`?

Reply:
`az group delete` deletes the Azure Resource Group whereas the expectation is to delete the Kubernetes cluster, namespaces and output folders created during the setup phase. We don't want to be creating a new resource group everytime we deploy a k8s cluster.

Reply:
Thanks @aashna!  I believe it's quite idiomatic to bundle all related ARM resources into a RG so they can be managed together? As far as I recall, `az group delete` guarantees that all related resources are deleted atomically in the correct order. Just my two cents.

Reply:
It's quite cumbersome to create an RG everytime you need to provision a new k8s cluster, so the idea was to not create it every time. But if your recommendation is to do that, I can change it to `az group delete`.